manifest/schema2/manifest.go

```diff
@@ -48,6 +52,14 @@ func init() {
 	}
 }
 
+// LayerDescriptor is an extension to distribution.Descriptor that
+// has additional fields specific to layers.
+type LayerDescriptor struct {
```

Comment:
I felt this way too initially, but @aaronlehmann suggested that we wanted to restrict this to layers.